### PR TITLE
Add utility functions for enhanced attrs and styles functionality

### PR DESCRIPTION
**Pull request by Snehadas2005**
Add new utility functions to improve developer experience when working with data attributes and CSS functions.

#### `attrs` package
- **`MergeDataAttrs`**: Create multiple data attributes from a map in one go
  ```go
  attrs.MergeDataAttrs(map[string]string{"id": "123", "role": "admin"})
  // Returns: Props{"data-id": "123", "data-role": "admin"}
  ```

#### `styles` package
- **`Calc`**: CSS calc() function wrapper
- **`Min`**: CSS min() function wrapper  
- **`Max`**: CSS max() function wrapper
- **`Clamp`**: CSS clamp() function for responsive values
- **`IsValidHexColor`**: Validate hex color format (#RGB or #RRGGBB)
- **`IsValidRGBColor`**: Validate RGB color format

### Benefits
- Simplifies working with multiple data attributes
- Type-safe CSS math functions for responsive design
- Built-in color validation for error prevention
- Maintains consistency with existing API patterns

### Testing
- Comprehensive unit tests for all new functions
- Edge case coverage for validators
- 100% test coverage for new code

### Documentation
- Function documentation with examples
- Updated README files for both packages

---

**I had worked on this contribution under the label of Hacktoberfest 2025.**

### CODE REVIEW

File: styles/utils.go [hunk 1]
```diff
@@ -155,3 +155,74 @@ func HSLA(hue, saturation, lightness int, alpha float64) string {
...
+	return "max(" + strings.Join(values, ", ") + ")"
+}
+
+// Clamp returns a CSS clamp() function string.
+func Clamp(min, preferred, max string) string {
+	return "clamp(" + min + ", " + preferred + ", " + max + ")"
+}
+
+// IsValidHexColor validates if a string is a valid hex color. Supports 3 or 6 digit hex codes.
+func IsValidHexColor(color string) bool {
```
[the remark this answers is not in the document]
Comment (chasefleming) [reply styles/utils.go]: I still see this in
